yellowbrick/classifier/boundaries.py

```diff
@@ -0,0 +1,457 @@
+
+        _, ncols = X_flat.shape
+
+        if ncols == 2:
+            X_two_cols = X
+            if self.features_ is None:
+                self.features_ = ["Feature One", "Feature Two"]
+
+        # Handle the feature names if they're None.
+        elif self.features_ is not None and is_dataframe(X):
+            X_two_cols = X[self.features_].as_matrix()
+
+        # handle numpy named/ structured array
+        elif self.features_ is not None and is_structured_array(X):
+            X_selected = X[self.features_]
+            X_two_cols = X_selected.view(np.float64).reshape(len(X_selected), -1)
+
+        # handle features that are numeric columns in ndarray matrix
+        elif self.features_ is not None and has_ndarray_int_columns(self.features_, X):
+            f_one, f_two = self.features_
+            X_two_cols = X[:, [int(f_one), int(f_two)]]
+
+        else:
+            raise YellowbrickValueError("""
+                ScatterVisualizer only accepts two features, please
+                explicitly set these two features in the init kwargs or
+                pass a matrix/ dataframe in with only two columns.""")
+
+        return X_two_cols
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        The fit method is the primary drawing input for the decision boundaries
+        visualization since it has both the X and y data required for the
+        viz and the transform method does not.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the visualizer
+        """
+        X = self._select_feature_columns(X)
+
+        # Assign each class a unique number for drawing
+        if self.classes_ is None:
+            self.classes_ = {
+                label: str(kls_num)
+                for kls_num, label in enumerate(np.unique(y))
+            }
+            self.class_labels = None
+        elif len(set(y)) == len(self.classes_):
+            self.classes_ = {
+                label: str(kls_num)
+                for kls_num, label in enumerate(self.classes_)
+            }
+            self.class_labels = dict(zip(set(y), self.classes_))
+        else:
+            raise YellowbrickTypeError(
+                """Number of classes must be the same length of number of
+                target y"""
+            )
+
+        # ensure that only
+        self.estimator.fit(X, y)
+
+        # Plot the decision boundary. For that, we will assign a color to each
+        # point in the mesh [x_min, x_max]x[y_min, y_max].
+        x_min, x_max = X[:, 0].min() - 1, X[:, 0].max() + 1
+        y_min, y_max = X[:, 1].min() - 1, X[:, 1].max() + 1
```

Comment:
So I get that you're trying to add some padding with the -1 and +1 on the min and the max. The problem is that if the range of feature is very small e.g. between 0.01 - 0.25, then adding or subtracting 1 here is going to result in a massive imbalance in the mesh. Also, though less bad, if the range of the feature is 10 to 1000, then the adding and subtracting of one will be barely visible. 

Potentially it might make sense to normalize X to the range [0, 1] -- this would solve the first problem. However, I might have a different suggestion depending on why you adjusted the range (perhaps the mesh should just go from min to max or be scaled by * 1.01 or something like that). Let me know!